src/core/StickyContainer.tsx

```diff
@@ -0,0 +1,160 @@
+/**
+ * Created by ananya.chandra on 14/09/18.
+ */
+
+import * as React from "react";
+import * as PropTypes from "prop-types";
+import {View} from "react-native";
+import RecyclerListView, {RecyclerListViewState, RecyclerListViewProps} from "./RecyclerListView";
+import { ScrollEvent } from "./scrollcomponent/BaseScrollView";
+import StickyObject, {StickyObjectProps, StickyObjectState} from "./sticky/StickyObject";
+import StickyHeader from "./sticky/StickyHeader";
+import StickyFooter from "./sticky/StickyFooter";
+import CustomError from "./exceptions/CustomError";
+import RecyclerListViewExceptions from "./exceptions/RecyclerListViewExceptions";
+
+export interface StickyContainerProps {
+    children: RecyclerChild;
+    stickyHeaderIndices?: number[];
+    stickyFooterIndices?: number[];
+    overrideRowRenderer?: (type: string | number | undefined, data: any, index: number, extendedState?: object) => JSX.Element | JSX.Element[] | null;
+}
+export interface StickyContainerState {
+    topVisible: boolean;
+}
+export interface RecyclerChild extends React.ReactElement<RecyclerListViewProps> {
+    ref: (recyclerRef: any) => {};
+    props: RecyclerListViewProps;
+}
+export default class StickyContainer<P extends StickyContainerProps, S extends StickyContainerState> extends React.Component<P, S> {
+    public static propTypes = {};
+    private _recyclerRef: RecyclerListView<RecyclerListViewProps, RecyclerListViewState> | null = null;
+    private _stickyHeaderRef: StickyObject<StickyObjectProps, StickyObjectState> | null = null;
+    private _stickyFooterRef: StickyObject<StickyObjectProps, StickyObjectState> | null = null;
+    private _initIndicesAll: number[] = [];
+    private _initIndicesNow: number[] = [];
+    private _initIndicesNotNow: number[] = [];
+
+    constructor(props: P, context?: any) {
+        super(props, context);
+        this._onVisibleIndicesChanged = this._onVisibleIndicesChanged.bind(this);
+        this._onScroll = this._onScroll.bind(this);
+
+        this.state = {
+            topVisible: false,
+        } as S;
+    }
+
+    public render(): JSX.Element {
+        this._assertChildType();
+        const recycler = React.cloneElement(this.props.children, {
+            ref: this._getRecyclerRef,
+            onVisibleIndicesChanged: this._onVisibleIndicesChanged,
+            onScroll: this._onScroll,
+        });
+        return (
+            <View style={{flex: 1}}>
```

Comment:
This will break web compatibility, or will it be RN only?
Allow style overriding

Reply:
Check for a case where RLV parent is larger and RLV has fixed height